terra_notebook_utils/utils.py

```diff
@@ -32,3 +32,14 @@ def __exit__(self, *args, **kwargs):
                 pass
             self.prune_futures()
         self._executor.shutdown()
+
+def concurrent_recursion(process_item: Callable[[Any], Iterable[Any]], items: Iterable[Any], concurrency: int=8):
```

Comment:
Nit: Two points:

1. I would expect a function with the name `recursion` to be a function calling itself and `concurrent_recursion` does not call `concurrent_recursion`.
2. This may be arguable, but I view recursion is an implementation detail.  Naming that describes the function's purpose rather than how it's implemented may be clearer and more robust against implementation changes in the future.

Reply:
I thought the functions _was_ named for it’s purpose (recursively calling `process_item`).

I’ll see if I can think of better names.